app/Jobs/UpdateWiki.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Libraries\OsuWiki;
+use App\Models\Wiki\Image;
+use App\Models\Wiki\Page;
+use App\Models\Wiki\WikiObject;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+
+class UpdateWiki implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable;
+
+    /** @var string */
+    private $oldHash;
+
+    /** @var string */
+    private $newHash;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param string $oldHash
+     * @param string $newHash
+     * @return void
+     */
+    public function __construct($oldHash, $newHash)
+    {
+        $this->oldHash = $oldHash;
+        $this->newHash = $newHash;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $files = OsuWiki::getUpdatedFiles($this->oldHash, $this->newHash);
+
+        foreach ($files as $file) {
+            $status = $file['status'];
+
+            $object = $this->getObject($file['filename']);
+
+            if ($status === 'added' || $status === 'renamed') {
+                // clearing out the cache for situations where something was cached
+                // even when there was no page in git, eg. an english page for
+                // a specific locale
+                $object->forget();
+
+                $object->get();
+            }
+
+            if ($status === 'renamed') {
+                $this->getObject($file['previous_filename'])->forget();
+            }
+
+            if ($status === 'removed' || $status === 'modified') {
+                $object->forget();
+            }
+
+            if ($status === 'modified') {
+                $object->get();
+            }
```

Comment:
this is a very long way to do

1. forget
2. forget previous_filename if renamed
3. get if not removed